chore: update extensions features

samuelmaddock opened the pull request:
#### Description of Change

Copied all non-private and non-chromeOS manifest and permission features from
- https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/extensions/api/_permission_features.json;drc=5691b7c58c31b1dd4e50056b2456050e6f92be86
- https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/extensions/api/_manifest_features.json;drc=5691b7c58c31b1dd4e50056b2456050e6f92be86

This will satisfy a [DCHECK when loading extensions](https://source.chromium.org/chromium/chromium/src/+/main:extensions/common/manifest_handlers/permissions_parser.cc;l=244-246;drc=c835ee197ea7448a3648187c8cd576bee9657d27) without a known permission. This also suppresses warnings when loading extensions.

cc @codebytere 

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation, tutorials, templates and examples are changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

codebytere reviewed (commented):
@samuelmaddock @jkleinsc I don't think we should blanket add API and manifest features we don't support. Given the permissions file DCHECKs I think that's more understandable but we didn't do that for API/manifest at the outset in order to make clearer via those APIs being `undefined` that we didn't support them.

samuelmaddock commented:
> to make clearer via those APIs being undefined that we didn't support them.

That's a good point. I can take a look at addressing your feedback next week and I'll open another PR. 👍
